src/Input/index.js

```diff
@@ -72,6 +81,8 @@ Input.propTypes = {
   startIconClass: PropTypes.node,
   /** full `narmi-icon-<shape>` className for icon at end of input */
   endIconClass: PropTypes.node,
+  displayClearInputIcon: PropTypes.bool,
+  clearInput: PropTypes.func,
```

Comment:
Should this logic be internal to `TextInput`, or is it important to give consumers direct control/responsibility over clearing behavior?

Reply:
I misread this as a prop on `TextInput` but it's actually on `Input`